src/jetstream.rs

```diff
@@ -780,7 +812,35 @@ impl JetStream {
 
         let _info: ConsumerInfo = self.js_request(&subject, &ser_req)?;
 
-        Consumer::existing::<&str, ConsumerConfig>(self.clone(), stream, config)
+        self.existing::<&str, ConsumerConfig>(&stream, config)
+    }
+
+    /// Instantiate a `JetStream` `Consumer`. Performs a check to see if the consumer
+    /// already exists, and creates it if not. If you want to use an existing
+    /// `Consumer` without this check and creation, use the `existing`
+    /// method.
+    pub fn create_or_open<S, C>(&self, stream: S, cfg: C) -> io::Result<Consumer>
```

Comment:
We use bind in other clients like the Go one.

Reply:
Renamed to `create_or_bind` to reflect that as `bind` does not try to create a consumer if it doesn't exist.